### ISSUE TITLE
docker-compose not working

### ISSUE THREAD

**Issue by Borkason** [docker/for-win#699]
### Expected behavior
docker-compose working as intended

### Actual behavior
`docker-compose` hangs. No output, no progress. Has to be killed with CTRL-C and then displays an error. Even on a simple `docker-compose --help`. Tested multiple times with stable and edge. Multiple restarts. Waited up to 15 minutes after running it before killing it with CTRL-C..

```
C:\Users\Borkason>docker-compose --version
Traceback (most recent call last):
  File "docker-compose", line 2, in <module>
  File "c:\users\appveyor\appdata\local\temp\1\pip-build-l5ncub\pyinstaller\PyInstaller\loader\pyimod03_importers.py", line 389, in load_module
  File "compose\cli\__init__.py", line 18, in <module>
  File "subprocess.py", line 479, in communicate
  File "subprocess.py", line 733, in _communicate
  File "threading.py", line 940, in join
  File "threading.py", line 340, in wait
KeyboardInterrupt
Failed to execute script docker-compose
```

### Information

The problem exists with Docker stable and Docker edge.
The diagnostic id for Docker edge is: 43C5302C-B344-4989-A4FD-783E62F2EA0B/2017-05-03_09-32-34

Windows 10, Version 1607, Build 14393.1066
Docker version 17.05.0-ce-rc1, build 2878a85

### Steps to reproduce the behavior

  1. install docker
  2. run `docker-compose --help`

**Comment by Borkason** [comment]
I have had this problem for weeks now, on two different computers running Windows 10.

Today, by pure chance, I started Kitematic and clicked on the button "DOCKER CLI" to see what would happen. Well, it started a powershell window. For no reason I tried running `docker-compose` and it was working. When opening a normal powershell or cmd window, `docker-compose` would still show the mentioned issues.

In short: docker-compose works when starting the docker-cli through kitematic.

**Comment by mylesbkeating1993** [comment]
When you are using a "normal powershell or cmd window", are you running as an administrator?

I've never tried Kitematic, but I'll explore a bit now and try to reproduce your issue

In the meantime, is running things through Kitematic an acceptable workaround or is there missing functionality?  Obviously this is an issue we want to solve either way - I just need to understand how much to prioritize working on this.

Cheers,
Myles

**Comment by Borkason** [comment]
> When you are using a "normal powershell or cmd window", are you running as an administrator?

Actually, I noticed that Kitematic opens a PowerShell as an administrator (despite asking for elevation, for whatever reason). There it works. In a normal PS it doesn't work.

Funny enough, when I tried the same thing at home, docker-compose suddenly works in a non-admin PowerShell. I know it didn't work some weeks ago, but let it drop because I had other things to do.

I run the same Docker version on both computers. My home computer however runs Windows 10, Version 1703, build 15063.250. That is not the only difference, though.

> In the meantime, is running things through Kitematic an acceptable workaround or is there missing functionality? Obviously this is an issue we want to solve either way - I just need to understand how much to prioritize working on this.

I can live with it. It's a bit annoying, but there is no functionality missing.

**Comment by mylesbkeating1993** [comment]
What is the scenario where you are trying to use docker-compose in a non-administrator Powershell session?

**Comment by Borkason** [comment]
My user account has no local admin privileges and I generally dislike having an administrative PowerShell open.

**Comment by mylesbkeating1993** [comment]
So with your user account, you are still able to run kinematic, which runs an elevated / administrator PS session?

I'm on Win10 Build 16170 with the same Docker version and can perform docker-compose commands in a non-admin PS session or command prompt.  ¯\_(ツ)_/¯

I can spin up a VM with your Windows later on.  In the meantime, do you have any Windows updates you haven't performed?

**Comment by Borkason** [comment]
> So with your user account, you are still able to run kinematic, which runs an elevated / administrator PS session?

No. I think when I shared a folder on C through Docker settings and entered local admin credentials, they were saved and used when I started Kitematic. After removing resetting the credentials in the Docker settings and restarting my computer, the CLI was not running under administrator anymore.

> I'm on Win10 Build 16170 with the same Docker version and can perform docker-compose commands in a non-admin PS session or command prompt.

Yes, same for me on my home computer. But at work (Active Directory environment) `docker-compose` does not work when running in a non-elevated command prompt, be it PowerShell or CMD.

Maybe the problem is not elevation, but being member of the local administrators group or not. Actually, I can evalute that... be right back.

**Comment by mylesbkeating1993** [comment]
Hi @Borkason 

Thanks for all the extra info and follow up!  Very helpful for me :)

It sounds like after your latest Docker update docker-compose is working great on all your machines, both at home and at work?  Do you know which update that was?

Please let me know if there are other issues, happy to help!

**Comment by Borkason** [comment]
Yeah, both.
Version 17.05.0-ce-win9 (11965)

Thanks for your support.

**Comment by mylesbkeating1993** [comment]
Sure, any time.  I'm going to close this now, but feel free to re-open if the same issue reappears. 

For new problems, opening a new issue is better for everyone.

Cheers,
M

**Comment by ffabinhoo** [comment]
docker-compose command is not working for me. I am running windows 10. When I run "docker-compose build" in a directory with the file docker-compose.yml, nothing happens. 
Anyone could help me? thanks

**Comment by gimoteco** [comment]
Same above happening with me, when I execute a command no error was printed, and cursor is available again. I'm using the version 18.06.1-ce-win73, my terminal is cmder, I also tried in cmd, git bash and the same effect previously described happend.
